src/main/java/Board.java

```diff
@@ -9,9 +12,74 @@ public Grid getGrid() {
         return grid;
     }
 
-    public void addMarkToGrid(int position, String mark) {
+    public void takeSquare(int position, String playerMark) {
         int index = position - 1;
-        grid.setSquare(index, mark);
+        grid.setSquare(index, playerMark);
+    }
+
+    public boolean moveValid
+            (int position, String player1Mark, String player2Mark) {
+
+        return positionValid(position)
+                && squareAvailable(position, player1Mark, player2Mark);
+    }
+
+    public boolean winningLine(String playerMark) {
+        var winner = false;
+        var count = 0;
+        var size = 0;
+
+        for (int[] combination : grid.getWinningCombinations()) {
+            size = combination.length;
+
+            for (int index : combination) {
+                if (grid.getSquare(index) == playerMark) count += 1;
+                if (count == size) winner = true;
+            }
+            count = 0;
+        }
+        return winner;
+    }
+
+    public boolean winningLineExists
+            (String player1Mark, String player2Mark) {
+        return winningLine(player1Mark) || winningLine(player2Mark);
+    }
+
+    public boolean full(String player1Mark, String player2Mark) {
+        var availableSquares = availableSquares(player1Mark, player2Mark);
+        return availableSquares.size() == 0;
```

Comment:
What do you think about keeping it all in one line?

Reply:
Sounds good! I'll change this